judge/views/problem.py

```diff
@@ -120,7 +120,9 @@ def get_context_data(self, **kwargs):
         solution = get_object_or_404(Solution, problem=self.object)
 
         if (not solution.is_public or solution.publish_on > timezone.now()) and \
-                not self.request.user.has_perm('judge.see_private_solution'):
+                not self.request.user.has_perm('judge.see_private_solution') or \
+                (self.request.user.is_authenticated and
+                 self.request.profile.current_contest):
```

Comment:
Can't this be on the same line and still be < 120 chars?

Reply:
It's 124 characters.